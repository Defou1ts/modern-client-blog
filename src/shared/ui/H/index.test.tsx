import React from 'react';

import { render, screen } from '@testing-library/react';

import renderer from 'react-test-renderer';

import { H } from './index';

describe('H component', () => {
	it('renders a H display', () => {
		render(<H type="display">Hello</H>);
		const text = screen.getByText('Hello');
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('renders a H1', () => {
		render(<H type="h1">Hello</H>);
		const text = screen.getByText('Hello');
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('renders a H2', () => {
		render(<H type="h2">Hello</H>);
		const text = screen.getByText('Hello');
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('renders a H3', () => {
		render(<H type="h3">Hello</H>);
		const text = screen.getByText('Hello');
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('renders a H4', () => {
		render(<H type="h4">Hello</H>);
		const text = screen.getByText('Hello');
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('renders a H5', () => {
		render(<H type="h5">Hello</H>);
		const text = screen.getByText('Hello');
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('renders a H6', () => {
		render(<H type="h6">Hello</H>);
		const text = screen.getByText('Hello');
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('H Display snapshot', () => {
		const component = renderer.create(<H type="display">Hello</H>);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('H1 snapshot', () => {
		const component = renderer.create(<H type="h1">Hello</H>);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('H2 snapshot', () => {
		const component = renderer.create(<H type="h2">Hello</H>);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('H3 snapshot', () => {
		const component = renderer.create(<H type="h3">Hello</H>);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('H4 snapshot', () => {
		const component = renderer.create(<H type="h4">Hello</H>);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('H5 snapshot', () => {
		const component = renderer.create(<H type="h5">Hello</H>);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('H6 snapshot', () => {
		const component = renderer.create(<H type="h6">Hello</H>);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
