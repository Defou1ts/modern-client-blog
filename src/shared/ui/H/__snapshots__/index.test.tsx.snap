// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H component H Display snapshot 1`] = `
<h1
  className="h display"
>
  Hello
</h1>
`;

exports[`H component H1 snapshot 1`] = `
<h1
  className="h h1"
>
  Hello
</h1>
`;

exports[`H component H2 snapshot 1`] = `
<h2
  className="h h2"
>
  Hello
</h2>
`;

exports[`H component H3 snapshot 1`] = `
<h3
  className="h h3"
>
  Hello
</h3>
`;

exports[`H component H4 snapshot 1`] = `
<h4
  className="h h4"
>
  Hello
</h4>
`;

exports[`H component H5 snapshot 1`] = `
<h5
  className="h h5"
>
  Hello
</h5>
`;

exports[`H component H6 snapshot 1`] = `
<h6
  className="h h6"
>
  Hello
</h6>
`;
